### PR TITLE
added "fix" option for autofixing style issues

### LINKED ISSUE #82: Support "auto-fixing" and "auto-configure" options
Not totally sure about them, since they might be out of scope of this plugin, but, why not?

### PR DESCRIPTION
**Pull request by spalger**
Fixes #82

### CODE REVIEW

**Comment by gustavohenke** [comment]
Nice work!
@markelog would you like to take a look?

All we'll have left to do is write docs for such feature.

File: tasks/lib/jscs.js [hunk 1]
```diff
@@ -58,16 +58,18 @@ exports.init = function( grunt ) {
     }
 
     /**
-     * @see Checker#checkPath
+     * @see Checker#checkPath and Checker#fixPath
      */
-    JSCS.prototype.check = function( path ) {
-        var checkPath = this.checker.checkPath( path );
+    JSCS.prototype.checkOrFix = function( path ) {
```
Comment (markelog) [inline tasks/lib/jscs.js]: Pretty polymorphic for my taste, but we can improve that later

**Comment by markelog** [comment]
@spalger could you add a documentation for it?

**Comment by spalger** [comment]
@markelog added. Not sure exactly what else to say.

File: tasks/lib/jscs.js (hunk 1 above)
Comment (spalger) [inline tasks/lib/jscs.js]: Agreed, would be happy to fix but this was the best I could come up with.

Comment (markelog) [inline tasks/lib/jscs.js]: Oh, okay, how about we add a method that return boolean value telling us if this option is enabled.

Then before the https://github.com/jscs-dev/grunt-jscs/pull/84/files#diff-7875685af27dacb398c11053ef46c5d0L21, we could call that method and depending of the return value, execute either, `check` or `fix` method.

How is that sound?

Comment (spalger) [inline tasks/lib/jscs.js]: Sounds good

**Comment by spalger** [comment]
@markelog moved to a `JSCS#check()` and `JSCS#fix()` approach

File: tasks/lib/jscs.js [hunk 2]
```diff
@@ -71,13 +71,26 @@ exports.init = function( grunt ) {
     };
 
     /**
+     * @see Checker#fixPath
+     */
+    JSCS.prototype.fix = function( path ) {
```
Comment (markelog) [inline tasks/lib/jscs.js]: It would be good to have a unit test for this method, not only a "enmasse" test

File: tasks/jscs.js [hunk 3]
```diff
@@ -13,12 +13,18 @@ module.exports = function( grunt ) {
...
                 // null is a default value, but its equivalent to `true`,
                 // with this way it's easy to distinguish specified value
-                config: null
+                config: null,
+
+                fix: false
             } ),
 
             jscs = new JSCS( options ),
             checks = this.filesSrc.map( function( path ) {
```
Comment (markelog) [inline tasks/jscs.js]: Looks good, although, we could do a small optimization here -

```js
this.filesSrc.map( jscs.getOptions().fix ? jscs.fix.bind( jscs ) : jscs.check.bind( jscs ) )
```
Or some variation of it, that way, that `if` would not be called on every iteration.

File: tasks/lib/jscs.js [hunk 4]
```diff
@@ -138,7 +151,7 @@ exports.init = function( grunt ) {
      * Get inline options
      * @return {Object}
      */
-    JSCS.prototype.getOptions = function() {
+    JSCS.prototype.getInlineOptions = function() {
```
Comment (markelog) [inline tasks/lib/jscs.js]: Strictly speaking, that would warrant major version

Comment (gustavohenke) [inline tasks/lib/jscs.js]: Do you know of someone who uses grunt-jscs as a lib for this to be a breaking change?

Comment (markelog) [inline tasks/lib/jscs.js]: We can and probably should consider this as internal API and therefore not bother ourself about semver in this context.

Anyway this is why i added "Strictly speaking"

File: tasks/lib/jscs.js (hunk 2 above)
Comment (spalger) [inline tasks/lib/jscs.js]: Added a unit test that makes sure it calls the checker. What else would you test?

**Comment by spalger** [comment]
Removed the changes to `JSCS#getOptions` since there is no reason for that code to manage or maintain the `fix` option. Now it now just filters the fix key from the options passed to jscs.

File: tasks/jscs.js [hunk 5]
```diff
@@ -13,13 +13,13 @@ module.exports = function( grunt ) {
...
+                config: null,
+
+                fix: false
             } ),
 
             jscs = new JSCS( options ),
-            checks = this.filesSrc.map( function( path ) {
-                return jscs.check( path );
-            } );
+            checks = this.filesSrc.map( options.fix ? jscs.fix : jscs.check, jscs );
```
Comment (markelog) [inline tasks/jscs.js]: cool

File: tasks/lib/jscs.js (hunk 2 above)
Comment (markelog) [inline tasks/lib/jscs.js]: I guess that would do the trick

**Comment by AurelioDeRosa** [comment]
Hi. Any idea of when this PR will be landed?

**Comment by markelog** [comment]
Doesn't really matter when it would be landed, it does matter when it would be released :-), i would like to release the new version with jscs 1.13, which should happen on this week

**Comment by gdborton** [comment]
Would love to see this land.  Any updated ETA on the 1.13 release?

**Comment by markelog** [comment]
Yep, today

**Pull request closed without merging**